pwd_guess.py

```diff
@@ -96,7 +102,7 @@ def pad_to_len(self, astring, maxlen=None):
     def encode_many(self, string_list, maxlen=None):
         maxlen = maxlen if maxlen else self.maxlen
         x_str_list = map(lambda x: self.pad_to_len(x, maxlen), string_list)
-        if self.embedding:
+        if self.sequence_model != Sequence.MANY_TO_MANY and self.embedding:
```

Comment:
Is there some reason for the extra check for the new model? It just seems like the new model also has configurable embedding, so I'm not sure why the check is here. I think I'm missing something, could you leave a comment about why?

Reply:
I'm looking into this, seems like the code path has become convoluted and I tried fixing it and it looks like I need to make design changes at several places.